src/pathpyG/core/MultiOrderModel.py

```diff
@@ -278,3 +282,280 @@ def from_DAGs(
+        dof_diff = self.get_mon_dof(max_order, assumption=assumption) - self.get_mon_dof(
+            max_order_null, assumption=assumption
+        )
+        # print(x, dof_diff)
+
+        # if the p-value is *below* the significance threshold, we reject the null hypothesis
+        p = 1 - chi2.cdf(x, dof_diff)
+        return (p < significance_threshold), p
+
+    def estimate_order(self, dag_data: DAGData, max_order: int = None, significance_threshold: float = 0.01) -> int:
+        """
+        Selects the optimal maximum order of a multi-order network model for the
+        observed paths, based on a likelihood ratio test with p-value threshold of p
+        By default, all orders up to the maximum order of the multi-order model will be tested.
+
+        @param paths: The path statistics for which to perform the order selection
+
+        @param maxOrder: The maximum order up to which the multi-order model shall be tested.
+        """
+        if max_order is None:
+            max_order = max(self.layers)  # THIS
+        assert max_order <= max(
+            self.layers
+        ), "Error: maxOrder cannot be larger than maximum order of multi-order network"
+        assert max_order > 1, "Error: max_order must be larger than one"
+
+        dag_graph = next(iter(DataLoader(dag_data.dags, batch_size=len(dag_data.dags)))).to(config["torch"]["device"])
+        assert set(dag_data.mapping.node_ids).intersection(set(self.layers[1].mapping.node_ids)) == set(dag_data.mapping.node_ids), "Input DAGData doesn t have the same set of nodes as those of the multi-order network"
+
+        max_accepted_order = 1
+
+        # Test for highest order that passes
+        # likelihood ratio test against null model
+        for k in range(2, max_order + 1):
+            if self.likelihood_ratio_test(
+                dag_graph, max_order_null=k - 1, max_order=k, significance_threshold=significance_threshold
+            )[0]:
+                max_accepted_order = k
+
+        return max_accepted_order
+
+def compute_weighted_outdegrees(graph: Graph) -> torch.Tensor:
+    """
+    Compute the weighted outdegrees of each node in the graph.
+
+    Args:
+        graph (Graph): pathpy graph object.
+
+    Returns:
+        tensor: Weighted outdegrees of nodes.
+    """
+    weighted_outdegree = scatter(
+        graph.data.edge_weight, graph.data.edge_index[0], dim=0, dim_size=graph.data.num_nodes, reduce="sum"
+    )
+    return weighted_outdegree
+
+def compute_transition_probabilities(graph: Graph) -> torch.Tensor:
+    """
+    Compute transition probabilities based on weighted outdegrees.
+
+    Args:
+        graph (Graph): pathpy graph object.
+
+    Returns:
+        tensor: Transition probabilities.
+    """
+    weighted_outdegree = compute_weighted_outdegrees(graph)
+    source_ids = graph.data.edge_index[0]
+    return graph.data.edge_weight / weighted_outdegree[source_ids]
```

Comment:
They might fit in the `centrality.py` in `algorithms.` But it could also be argued that they should become methods of `Graph` object. Should I put them in `centrality.py` and potentially we'll move them in the future?

Reply:
Adding them as methods to `Graph` sounds like a good idea to me. 👍🏼

Reply:
I have moved those methods to `Graph`